src/Providers/AuthUserProvider.tsx

```diff
@@ -42,23 +42,28 @@ export default function AuthUserProvider({ children, unauthorized }: Props) {
         localStorage.setItem(LocalStorageKeys.refreshToken, query.data.refresh);
 
         await refetch();
-        navigate(getRedirectOr("/"));
+        navigate(getRedirectOr(location.pathname));
       }
 
       return query;
     },
     [refetch]
   );
 
-  const signOut = useCallback(async () => {
-    localStorage.removeItem(LocalStorageKeys.accessToken);
-    localStorage.removeItem(LocalStorageKeys.refreshToken);
-
-    await refetch();
-
-    const redirectURL = getRedirectURL();
-    navigate(redirectURL ? `/?redirect=${redirectURL}` : "/");
-  }, [refetch]);
+  const signOut = useCallback(
+    async (noRedirect = false) => {
+      localStorage.removeItem(LocalStorageKeys.accessToken);
+      localStorage.removeItem(LocalStorageKeys.refreshToken);
+
+      await refetch();
+      navigate(
+        getRedirectOr(
+          noRedirect ? "/login" : `/login?redirect=${location.href}`
```

Comment:
Instead of directly appending `location.href` to set the redirect, let's URL encode it first, this will preserve the query parameters of the target page. When visiting the redirection, we can URL decode this `redirect` paramater to get the actual URL. If we do not URL encode, like it is right now, then during redirection we are losing the query parameters.

Instead of `/login?redirect=https://example.com/search?q1=test&q2=test2`, it will appear as `https%3A%2F%2Fexample.com%2Fsearch%3Fq1%3Dtest%26q2%3Dtest2`

Now when reading the redirect parameter, in former case it will only return example.com/search but the latter would return the full url even with the q1 and q2 query params.

cc: @rithviknishad to comment and confirm on this approach

Reply:
I agree with @Ashesh3

Reply:
Fixed with the latest commit.